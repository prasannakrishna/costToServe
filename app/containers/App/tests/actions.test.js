/**
 * Copyright Â© 2018, JDA Software Group, Inc. ALL RIGHTS RESERVED.
 * <p>
 * This software is the confidential information of JDA Software, Inc., and is licensed
 * as restricted rights software. The use,reproduction, or disclosure of this software
 * is subject to restrictions set forth in your license agreement with JDA.
 */
import {
  loadRepos,
  reposLoaded,
  repoLoadingError,
  logout,
  getAlertsCount,
  alertsCountLoaded,
  alertsCountError,
  getMetaData,
  metaDataLoaded,
  metaDataLoadingError,
  getBusinessProcesses,
  businessProcessesLoaded,
  businessProcessesLoadingError,
} from '../actions';

describe('App Actions', () => {
  describe('loadRepos Action', () => {
    it('has a type of LOAD_REPOS', () => {
      expect(loadRepos()).toMatchSnapshot();
    });
  });
  describe('reposLoaded Action', () => {
    it('has a type of LOAD_REPOS_SUCCESS', () => {
      expect(reposLoaded()).toMatchSnapshot();
    });
  });
  describe('repoLoadingError Action', () => {
    it('has a type of LOAD_REPOS_ERROR', () => {
      expect(repoLoadingError()).toMatchSnapshot();
    });
  });
  describe('logout Action', () => {
    it('has a type of LOGOUT', () => {
      expect(logout()).toMatchSnapshot();
    });
  });
  describe('getAlertsCount Action', () => {
    it('has a type of LOAD_ALERTS_COUNT', () => {
      expect(getAlertsCount()).toMatchSnapshot();
    });
  });
  describe('alertsCountLoaded Action', () => {
    it('has a type of LOAD_ALERTS_COUNT_SUCCESS', () => {
      expect(alertsCountLoaded()).toMatchSnapshot();
    });
  });
  describe('alertsCountError Action', () => {
    it('has a type of LOAD_ALERTS_COUNT_ERROR', () => {
      expect(alertsCountError()).toMatchSnapshot();
    });
  });
  describe('getMetaData Action', () => {
    it('has a type of LOAD_METADATA', () => {
      expect(getMetaData()).toMatchSnapshot();
    });
  });
  describe('metaDataLoaded Action', () => {
    it('has a type of LOAD_METADATA_SUCCESS', () => {
      expect(metaDataLoaded()).toMatchSnapshot();
    });
  });
  describe('metaDataLoadingError Action', () => {
    it('has a type of LOAD_METADATA_ERROR', () => {
      expect(metaDataLoadingError()).toMatchSnapshot();
    });
  });
  describe('getBusinessProcesses Action', () => {
    it('has a type of LOAD_BUSINESS_PROCESSES', () => {
      expect(getBusinessProcesses()).toMatchSnapshot();
    });
  });
  describe('businessProcessesLoaded Action', () => {
    it('has a type of LOAD_BUSINESS_PROCESSES_SUCCESS', () => {
      expect(businessProcessesLoaded()).toMatchSnapshot();
    });
  });
  describe('businessProcessesLoadingError Action', () => {
    it('has a type of LOAD_BUSINESS_PROCESSES_ERROR', () => {
      expect(businessProcessesLoadingError()).toMatchSnapshot();
    });
  });
});
