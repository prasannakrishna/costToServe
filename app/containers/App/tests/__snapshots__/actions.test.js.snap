// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Actions alertsCountError Action has a type of LOAD_ALERTS_COUNT_ERROR 1`] = `
Object {
  "error": undefined,
  "type": "boilerplate/App/LOAD_ALERTS_COUNT_ERROR",
}
`;

exports[`App Actions alertsCountLoaded Action has a type of LOAD_ALERTS_COUNT_SUCCESS 1`] = `
Object {
  "alertsCount": undefined,
  "type": "boilerplate/App/LOAD_ALERTS_COUNT_SUCCESS",
}
`;

exports[`App Actions businessProcessesLoaded Action has a type of LOAD_BUSINESS_PROCESSES_SUCCESS 1`] = `
Object {
  "businessProcesses": undefined,
  "type": "boilerplate/App/LOAD_BUSINESS_PROCESSES_SUCCESS",
}
`;

exports[`App Actions businessProcessesLoadingError Action has a type of LOAD_BUSINESS_PROCESSES_ERROR 1`] = `
Object {
  "error": undefined,
  "type": "boilerplate/App/LOAD_BUSINESS_PROCESSES_ERROR",
}
`;

exports[`App Actions getAlertsCount Action has a type of LOAD_ALERTS_COUNT 1`] = `
Object {
  "durationObj": undefined,
  "filterId": undefined,
  "preferences": undefined,
  "type": "boilerplate/App/LOAD_ALERTS_COUNT",
}
`;

exports[`App Actions getBusinessProcesses Action has a type of LOAD_BUSINESS_PROCESSES 1`] = `
Object {
  "type": "boilerplate/App/LOAD_BUSINESS_PROCESSES",
}
`;

exports[`App Actions getMetaData Action has a type of LOAD_METADATA 1`] = `
Object {
  "type": "boilerplate/App/LOAD_METADATA",
}
`;

exports[`App Actions loadRepos Action has a type of LOAD_REPOS 1`] = `
Object {
  "type": "boilerplate/App/LOAD_REPOS",
}
`;

exports[`App Actions logout Action has a type of LOGOUT 1`] = `
Object {
  "type": "boilerplate/App/LOGOUT",
}
`;

exports[`App Actions metaDataLoaded Action has a type of LOAD_METADATA_SUCCESS 1`] = `
Object {
  "metaData": undefined,
  "type": "boilerplate/App/LOAD_METADATA_SUCCESS",
}
`;

exports[`App Actions metaDataLoadingError Action has a type of LOAD_METADATA_ERROR 1`] = `
Object {
  "error": undefined,
  "type": "boilerplate/App/LOAD_METADATA_ERROR",
}
`;

exports[`App Actions repoLoadingError Action has a type of LOAD_REPOS_ERROR 1`] = `
Object {
  "error": undefined,
  "type": "boilerplate/App/LOAD_REPOS_ERROR",
}
`;

exports[`App Actions reposLoaded Action has a type of LOAD_REPOS_SUCCESS 1`] = `
Object {
  "repos": undefined,
  "type": "boilerplate/App/LOAD_REPOS_SUCCESS",
  "username": undefined,
}
`;
